compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -975,6 +997,63 @@ fn can_be_hoisted(
     }
 }
 
+/// Indicates if the instruction has a side effect, ie. it can fail, or it interacts with memory.
+fn has_side_effects(instruction: &Instruction, dfg: &DataFlowGraph) -> bool {
```

Comment:
I guess this is the same as the one in constant_folding, right? If so, maybe it could be moved to `Instruction`. It was previously moved here because it was the only place it was used, but now it could maybe move back to where it was.

Reply:
Yeah it is the same. I was unsure on whether we wanted to keep these methods separate still and just duplicate the code within the pass. The semantics of a side-effectual instructions shouldn't matter per pass like deduplicated vs. hoisting though, so I can de-duplicate this method.